itests/mir_test.go

```diff
@@ -231,6 +240,78 @@ func (ts *itestsConsensusSuite) testMirWhenLearnersJoin(t *testing.T) {
 	require.NoError(t, err)
 }
 
+// testMirWithReconfiguration tests that the reconfiguration mechanism operates normally
+// if a new validator joins the network.
+func (ts *itestsConsensusSuite) testMirWithReconfiguration(t *testing.T) {
+	var wg sync.WaitGroup
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer func() {
+		t.Logf("[*] defer: cancelling %s context", t.Name())
+		cancel()
+		wg.Wait()
+	}()
+
+	membershipFileName := "_membership_config.tmp"
+	t.Cleanup(func() {
+		os.Remove(membershipFileName) // nolint
+	})
+
+	nodes, miners, ens := kit.EnsembleMirNodes(t, MirTotalValidatorNumber+1, ts.opts...)
+	ens.StoreMirValidatorsToMembersipFile(membershipFileName, miners[:MirTotalValidatorNumber]...)
+	ens.InterconnectFullNodes().BeginMirMiningWithMembershipFromFile(ctx, membershipFileName, &wg, 0, miners[:MirTotalValidatorNumber])
+
+	err := kit.AdvanceChain(ctx, TestedBlockNumber, nodes[:MirTotalValidatorNumber]...)
+	require.NoError(t, err)
+	err = kit.CheckNodesInSync(ctx, 0, nodes[0], nodes[1:MirTotalValidatorNumber]...)
+	require.NoError(t, err)
+
+	t.Log(">>> new validators have been added to the membership")
+	ens.StoreMirValidatorsToMembersipFile(membershipFileName, miners...)
+	// Start new miners.
+	ens.InterconnectFullNodes().BeginMirMiningWithMembershipFromFile(ctx, membershipFileName, &wg, 0, miners[MirTotalValidatorNumber:])
+
+	err = kit.AdvanceChain(ctx, TestedBlockNumber, nodes...)
+	require.NoError(t, err)
+	err = kit.CheckNodesInSync(ctx, 0, nodes[0], nodes...)
+	require.NoError(t, err)
+}
```

Comment:
Can we also add here an additional reconfiguration window or an additional test where we test removing a validator from the set?